### PR TITLE
GT6 style wires - Solar panel(cover) fix

### PR DESCRIPTION
**Pull request by mitchej123**
Make solar panels always look connected, and allow energy injection from
unknown directions

### CODE REVIEW

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaPipeEntity_Cable.java [hunk 1]
```diff
@@ -341,35 +345,67 @@ public int connect(byte aSide) {
...
+            if (tTileEntity instanceof IColoredTileEntity) {
+                if (getBaseMetaTileEntity().getColorization() >= 0) {
+                    byte tColor = ((IColoredTileEntity) tTileEntity).getColorization();
+                    if (tColor >= 0 && tColor != getBaseMetaTileEntity().getColorization()) {
+                    	return rConnect;
+                    }
+                }
+            }
+            
+            boolean sHasSolarPanel = coverBehavior instanceof GT_Cover_SolarPanel;
```
Comment (mitchej123) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaPipeEntity_Cable.java]: I don't love this; but I'm not aware of any other "covers" that inject energy.  Also it's the only connection that doesn't require a tile entity on the other side.

**Comment by Dream-Master** [comment]
@mitchej123 thanks

**Comment by Technus** [comment]
That could also be addressed
(should allow energy in on solar cover = false?)
image
Just so you cannot connect solar panel from top (the other side)

**Comment by Technus** [comment]
Also impossible to disconnect
image

**Comment by mitchej123** [comment]
**Re: Impossible to Disconnect** -- This happens with anything currently that's marked as always show connected (pump covers too).  It's especially annoying with large pipes that block being able to screwdriver a pump on a machine -- was going to look into this separately.  This is also consistent with previous behavior so I wasn't particularly worried about it immediately.

**Re: Allow energy in/out** - I think this is already set to false (which is why I set always connected = true); It lets energy pass through, it doesn't address a cover producing energy.  The top wire will show being connected, but I don't think it'll pass anything through.    One option is for "always connected" but not "let energy in/out" only connect the cover to the entity it's covering.

**re:transferElectricity** - All of the injectEnergy calls from solar panels (and a few other places) use transferElectricity with an UNKNOWN side, so both need to handle it.  There are no direct invocations of transferElectricity with an UNKNOWN side that don't go through injectEnergy:

```
$ egrep -l "injectEnergyUnits(.*)6" src/* --recursive
src/main/java/gregtech/common/tileentities/machines/basic/GT_MetaTileEntity_MicrowaveEnergyTransmitter.java
src/main/java/gregtech/common/covers/GT_Cover_SolarPanel.java
src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Transformer.java

$ egrep -l "transferElectricity(.*)6" src/* --recursive
$
```

**Pull request merged**